lms/static/scripts/frontend_apps/config.ts

```diff
@@ -205,28 +205,40 @@ export type Product = {
  * the backend code.
  */
 export type ConfigObject = {
+  // Configuration present in all modes.
   mode: AppMode;
   api: {
     authToken: string;
-    sync: APICallInfo;
-    viaUrl: APICallInfo;
+
+    // Only present in "basic-lti-launch" mode.
+    sync?: APICallInfo;
+    viaUrl?: APICallInfo;
   };
+  dev: boolean;
+  debug?: DebugInfo;
+  product: Product;
+
+  // Only present in "basic-lti-launch" mode.
   canvas: {
-    speedGrader: SpeedGraderConfig;
+    // Only present in Canvas.
+    speedGrader?: SpeedGraderConfig;
   };
   contentBanner?: ContentBannerConfig;
-  dev: boolean;
-  filePicker: FilePickerConfig;
-  grading: GradingConfig;
+  grading?: GradingConfig; // Only present for instructors
   hypothesisClient: ClientConfig;
   rpcServer: {
     allowedOrigins: string[];
   };
   viaUrl: string;
-  OAuth2RedirectError: OAuthErrorConfig;
-  errorDialog: ErrorDialogConfig;
-  debug?: DebugInfo;
-  product: Product;
+
+  // Only present in "error-dialog" mode.
+  errorDialog?: ErrorDialogConfig;
+
+  // Only present in "content-item-selection" mode.
+  filePicker: FilePickerConfig;
```

Comment:
The optionality here is inconsistent with the other mode-specific objects. I will address that separately.